New autoconf gets upset if AC_PROC_CC and friends aren't called at
the top level (see http://www.gnu.org/software/hello/manual/autoconf/Expanded-Before-Required.html)

Fix macros accordingly fixing build failures

RP 2/2/10

Upstream-Status: Inappropriate [configuration]

Index: taglib-1.6.1/acinclude.m4
===================================================================
--- taglib-1.6.1.orig/acinclude.m4	2010-02-01 23:57:16.857788731 +0000
+++ taglib-1.6.1/acinclude.m4	2010-02-01 23:58:08.239039503 +0000
@@ -3141,12 +3141,7 @@
     [kde_use_profiling="no"]
   )
 
-  dnl this prevents stupid AC_PROG_CC to add "-g" to the default CFLAGS
-  CFLAGS=" $CFLAGS"
 
-  AC_PROG_CC 
-
-  AC_PROG_CPP
 
   if test "$GCC" = "yes"; then
     if test "$kde_use_debug_code" != "no"; then
@@ -3174,9 +3169,6 @@
      LDFLAGS=""
   fi
 
-  CXXFLAGS=" $CXXFLAGS"
-
-  AC_PROG_CXX
 
   KDE_CHECK_FOR_BAD_COMPILER
 
Index: taglib-1.6.1/configure.in
===================================================================
--- taglib-1.6.1.orig/configure.in	2010-02-01 23:46:47.389039146 +0000
+++ taglib-1.6.1/configure.in	2010-02-01 23:58:14.317819479 +0000
@@ -36,6 +36,21 @@
 dnl This ksh/zsh feature conflicts with `cd blah ; pwd`
 unset CDPATH
 
+dnl this prevents stupid AC_PROG_CC to add "-g" to the default CFLAGS
+CFLAGS=" $CFLAGS"
+
+AC_PROG_CC 
+
+AC_PROG_CPP
+
+CXXFLAGS=" $CXXFLAGS"
+
+AC_PROG_CXX
+
+dnl checks for programs.
+dnl first check for c/c++ compilers
+AC_CHECK_COMPILERS
+
 dnl Checking host/target/build systems, for make, install etc.
 AC_CANONICAL_SYSTEM
 dnl Perform program name transformation
@@ -62,9 +77,7 @@
 AM_CONFIG_HEADER(config.h)
 AM_CONFIG_HEADER(taglib/taglib_config.h)
 
-dnl checks for programs.
-dnl first check for c/c++ compilers
-AC_CHECK_COMPILERS
+
 
 dnl CXXFLAGS="$NOOPT_CXXFLAGS" dnl __kdevelop[noopt]__
 dnl CFLAGS="$NOOPT_CFLAGS" dnl __kdevelop[noopt]__
